sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -136,6 +136,30 @@ object UnresolvedTableValuedFunction {
   }
 }
 
+/**
+ * A table-valued function with output column aliases. The table function has been
+ * looked up and turned into a logical plan.
+ *
+ * @param name qualified name of the table-valued function
+ * @param child logical plan of the table-valued function
+ * @param outputNames alias names of function output columns. The analyzer adds [[Project]]
+ *                    to rename the output columns.
+ */
+case class TableValuedFunctionWithAlias(
```

Comment:
I think the confusing part is, `UnresolvedTableValuedFunction` contains alias as well. If we add a new node `UnresolvedTVFAlias`, this could be clearer:
- parser creates `UnresolvedTVFAlias(UnresolvedTableValuedFunction)`
- analyzer turns `UnresolvedTableValuedFunction` into resolved TVF plan
- analyzer turns `UnresolvedTVFAlias` into `Project` if its child is resolved.

Reply:
+1 for @cloud-fan 's suggestion.